keras_cv/utils/fill_utils.py

```diff
@@ -33,24 +33,28 @@ def rectangle_masks(corners, mask_shape):
     corners = corners[..., tf.newaxis, tf.newaxis]
 
     # split coordinates
-    x0 = corners[:, 0]
-    y0 = corners[:, 1]
-    x1 = corners[:, 2]
-    y1 = corners[:, 3]
+    x0 = corners[:, :, 0]
+    y0 = corners[:, :, 1]
+    x1 = corners[:, :, 2]
+    y1 = corners[:, :, 3]
 
     # repeat height and width
     width, height = mask_shape
-    x0_rep = tf.repeat(x0, height, axis=1)
-    y0_rep = tf.repeat(y0, width, axis=2)
-    x1_rep = tf.repeat(x1, height, axis=1)
-    y1_rep = tf.repeat(y1, width, axis=2)
+    x0_rep = tf.repeat(x0, height, axis=2)
+    y0_rep = tf.repeat(y0, width, axis=3)
```

Comment:
can we maybe use negative indexes here so that the number of dimensions in `x0` and `y0` etc don't matter?